src/logic/safe/transactions/awaitingTransactions.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+import { List } from 'immutable'
+import type { Transaction } from '~/routes/safe/store/models/transaction'
+
+export const getAwaitingTransactions = (allTransactions: List<Transaction>, userAccount: string, userIsOwner: boolean): List<Transaction> => {
+  if (!allTransactions || !userIsOwner) {
+    return List([])
+  }
+
+  const allAwaitingTransactions = allTransactions.map((safeTransactions) => {
+    const nonCancelledTransactions = safeTransactions.filter((transaction: Transaction) => {
+      // If transactions is not executed, but there's a transaction with the same nonce submitted later
```

Comment:
Actually this is not the case anymore, after a recent change transaction is cancelled only if transaction with the same nonce was executed (not submitted). Sorry for not communicating this

Reply:
Could you add more details to this? I didn't get it and I'm not sure about what to change here

Reply:
> but there's a transaction with the same nonce submitted later it means that the transaction was cancelled (Replaced) and shouldn't get executed

I'm talking about this comment. Actually, maybe it's a little bit early, because https://github.com/gnosis/safe-react/pull/293 is not merged yet. So what I'm saying is that after this PR is merged if you submit a cancellation tx with the same nonce you still can execute either of those two (cancellation and the one you wanted to cancel)

Reply:
Got it now, I added #293 as depency of this pr and will change that logic

Reply:
I changed  the logic, now I recognize a canceled transaction if there is another tx with the same nonce and it's executed, let me know if this logic it's ok please. Thanks!